src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/SortClauseDeserializer.java

```diff
@@ -51,36 +52,69 @@ public SortClause deserialize(JsonParser parser, DeserializationContext ctxt) th
     while (fieldIter.hasNext()) {
       Map.Entry<String, JsonNode> inner = fieldIter.next();
       String path = inner.getKey().trim();
-
+      float[] arrayVals = null;
+      if (inner.getValue().isObject()) {
+        var ejsonWrapped = EJSONWrapper.maybeFrom((ObjectNode) inner.getValue());
+        if (ejsonWrapped == null) {
+          throw ErrorCodeV1.INVALID_SORT_CLAUSE_VALUE.toApiException(
+              "Only binary vector object values is supported for sorting. Path: %s, Value: %s.",
+              path, inner.getValue().toString());
+        }
+        try {
+          arrayVals = ejsonWrapped.getVectorValueForBinary();
+        } catch (RuntimeException e) {
```

Comment:
Similar to the other case, would we only expect `IllegalArgumentException`? If so, catch just that.